File: backend/entity/Employee.go
package entity

import ("gorm.io/gorm"
"time")

type Employee struct {
	gorm.Model
	EmployeeID   uint      `json:"employeeID" gorm:"primaryKey;autoIncrement"`
	ProfileImage string    `json:"profileImage"`
	FirstName    string    `json:"firstName"`
	LastName     string    `json:"lastName"`
	Password     string    `json:"-"`
	Email        string    `json:"email" gorm:"uniqueIndex"`
	Phone        string    `json:"phone"`
	Address      string    `json:"address"`
	Birthday     time.Time `json:"birthday"`
	Sex          string    `json:"sex"`
	Position     string    `json:"position"`
	JobType      string    `json:"jobType"`
	TotalSales   string    `json:"totalSales"`

	LeaveRequests  []LeaveRequest   `json:"leaves" gorm:"-"`
	PickupDelivery []PickupDelivery `gorm:"foreignKey:EmployeeID"`
	Car            []Car            `gorm:"foreignKey:EmployeeID"`
	SaleList       []SaleList       `gorm:"foreignKey:EmployeeID"`
	SalesContract  []SalesContract  `gorm:"foreignKey:EmployeeID"`
}

File: frontend/src/types/employee.ts
export type Sex = "" | "male" | "female" | "other";
export type JobType = "" | "Full-time" | "Part-time" | "Contract";

export interface Employee {
  employeeID: number;
  profileImage: string;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  address: string;
  sex: Sex;
  birthday?: string;  // ✅ ใช้ string (YYYY-MM-DD)
  position: string;
  jobType: JobType;
  totalSales: string;
}
